### PR TITLE
[REEF-545]: Create a Configuration Helper for NetworkConnectionService

### PR DESCRIPTION
**Pull request by taegeonum**
This pull request addressed the issue by
* creating `BindNetworkConnectionServiceToDriver` which registers driver id to `NetworkConnectionService`.
* creating `NetworkConnectionServiceConfiguration` which provides driver configuration and service configuration.

JIRA: [REEF-545](https://issues.apache.org/jira/browse/REEF-545)

### CODE REVIEW

File: lang/java/reef-io/src/main/java/org/apache/reef/io/network/impl/UnbindNetworkConnectionServiceFromTask.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 import javax.inject.Inject;
 /**
  * TaskStop event handler for unregistering NetworkConnectionService.
- * Users have to bind this handler into ServiceConfiguration.ON_TASK_STOP.
+ * This class unregisters driverId from network connection service.
```
Comment (dafrista) [inline lang/java/reef-io/src/main/java/org/apache/reef/io/network/impl/UnbindNetworkConnectionServiceFromTask.java]: Typo: `driverId` -> `taskId`

**Comment by dafrista** [comment]
This PR assumes that NCS will be used from a Task. Other services may want to use NCS as well. Does adding a configuration helper for Service/Context make sense at all?

**Comment by taegeonum** [comment]
@dafrista Good question! I just supposed that services which use NCS run on top of a task. That said, NCS is a service for communication between tasks. I didn't consider a situation where NCS is used in contexts. This is why current NCS has just one id. 

Maybe, we need to support multiple ids per NCS if we want the NCS is also used in multiple contexts.

**Comment by bgchun** [comment]
We should not assume that the NCS is used only by tasks.

On Tue, Aug 4, 2015 at 3:44 PM, Tae-Geon Um <notifications@github.com>
wrote:

> @dafrista <https://github.com/dafrista> Good question! I just supposed
> that services which use NCS run on top of a task. That said, NCS is a
> service for communication between tasks. I didn't consider a situation
> where NCS is used in contexts. This is why current NCS has just one id.
>
> Maybe, we need to support multiple ids per NCS if we want the NCS is also
> used in multiple contexts.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/329#issuecomment-127498600>
> .
>



-- 
Byung-Gon Chun

**Comment by taegeonum** [comment]
@dafrista @bgchun Actually, NCS can be used within contexts if the client services use taskId for communication. But, it is not possible to use NCS before task is started. Is this a problem? and Is it reasonable for a NCS to have multiple ids?

**Comment by taegeonum** [comment]
@dafrista @bgchun  Sorry for the confusion! The NCS can be used within contexts if the services use taskId for communication. That said, other services can use NCS as well. Using just one id per NCS looks good to me. What do you think?

**Comment by taegeonum** [comment]
I agree that NCS should keep multiple identifiers to support services which are used in contexts. This pull request does not support that feature. I will close this issue.

**Pull request closed without merging**